app/src/main/java/com/framgia/soundclound/screen/detailalbum/DetailAlbumModelView.java

```diff
@@ -44,21 +52,56 @@ public void setDetailAlbumAdapter(DetailAlbumAdapter detailAlbumAdapter) {
         notifyPropertyChanged(0);
     }
 
+    public void onUpdate() {
+        mTracks = AlbumRepository.getInstance(mContext).getAllTrack(mIdAlbum);
+        mDetailAlbumAdapter.updateData(mTracks);
+    }
+
     @Override
     public void onItemClick(Track track, int pos) {
-        // // TODO: 17/1/2018 save data SharePreferences
+        SharePreferences.getInstance().putListTrack(new Gson().toJson(mTracks));
+        SharePreferences.getInstance().putTrack(new Gson().toJson(track));
+        SharePreferences.getInstance().putIndex(pos);
         mContext.startActivity(PlayTrackActivity.getInstance(mContext));
     }
 
     public void onFabAddAlbumClick() {
-        mContext.startActivity(AddTrackActivity.getInstance(mContext, mIdAlbum));
+        Intent intent = AddTrackActivity.getInstance(mContext, mIdAlbum, AddTrackActivity.KEY_ADD_TRACK_ALBUM);
+        ((Activity) mContext).startActivityForResult(intent, DetailAlbumActivity.REQUEST_UPDATE_DATA);
     }
 
     @Override
     public void onIconMoreClick(ImageView imageView, final Track track) {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
```

Comment:
ERROR: [STYLE][DLS_DEAD_LOCAL_STORE] Dead store to builder in com.framgia.soundclound.screen.detailalbum.DetailAlbumModelView.onIconMoreClick(ImageView, Track)
http://findbugs.sourceforge.net/bugDescriptions.html#DLS_DEAD_LOCAL_STORE